src/shadowkeep/lib/coordinates.py

```diff
@@ -0,0 +1,70 @@
+class Coordinates(list):
+    def __init__(self, x=0, y=0):
+        super().__init__([x, y])
+
+    @property
+    def x(self):
+        return self[0]
+
+    @x.setter
+    def x(self, value):
+        self[0] = value
+
+    @property
+    def y(self):
+        return self[1]
+
+    @y.setter
+    def y(self, value):
+        self[1] = value
+
+    def __add__(self, other):
+        return self.x + other.x, self.y + other.y
+
+    def __iadd__(self, other):
+        self.x += other.x
+        self.y += other.y
+        return self
+
+    def __str__(self):
+        return f"Coordinates({self.x}, {self.y})"
+
+    def is_neighbour(self, other):
+        return (
+            abs(self.x - other.x) == 1
+            and abs(self.y - other.y) == 0
```

Comment:
`abs(self.y - other.y) == 0` bych zapsal spíš jen jako `self.y == other.y` :)
(to samé pro .x)